sql3/planner/expression.go

```diff
@@ -1998,6 +1998,48 @@ func (n *boolLiteralPlanExpression) WithChildren(children ...types.PlanExpressio
 	return n, nil
 }
 
+// represents date system variable such as CURRENT_DATE and CURRENT_DATETIME
+type dateSysVariablePlanExpression struct {
+	name  string
+	value time.Time
+}
+
+func newDateSysVariablePlanExpression(name string, value time.Time) *dateSysVariablePlanExpression {
+	return &dateSysVariablePlanExpression{
+		name:  name,
+		value: value,
+	}
+}
+
+func (n *dateSysVariablePlanExpression) Evaluate(currentRow []interface{}) (interface{}, error) {
```

Comment:
Why pre-load the value somewhere else? Why not just evaluate here? 

Also consider a future where compilation and execution are separated in time - e.g. create view compiles and stores a binary representation of an execution plan which gets executed some time later.